scalalib/src/mill/scalalib/JavaModule.scala

```diff
@@ -424,8 +427,7 @@ trait JavaModule
    * necessary to run this module's code after compilation
    */
   def runClasspath: T[Seq[PathRef]] = T {
-    localClasspath() ++
-      upstreamAssemblyClasspath()
+    resolvedRunIvyDeps().toSeq ++ transitiveLocalClasspath() ++ localClasspath()
```

Comment:
We remove `upstreamAssemblyClasspath` so we can remove the dependency on `unmanagedClasspath`, since it now also comes from `localClasspath`, and we want to avoid duplicate classpath entries